### PR TITLE
KEB configuration for given Kyma version and plan

### PR DESCRIPTION
**Pull request by szwedm**
**Description**

Changes proposed in this pull request:

- add configuration provider for given runtime version and plan,
- refactor ComponentsProvider.AllComponents() to accept ConfigForPlan struct,
- add/refactor tests.

**Related issue(s)**
See #1824

### CODE REVIEW

File: components/kyma-environment-broker/internal/config/reader_test.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+		ObjectMeta: metaV1.ObjectMeta{
+			Name:      m.Metadata.Name,
+			Namespace: m.Metadata.Namespace,
+			Labels:    m.Metadata.Labels,
+		},
+		Data: m.Data,
+	}
+}
+
+type mockK8sClient struct{}
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/config/reader_test.go:199]: I'd name it like: "failingK8sClient" or something like this which tells us it always fails

File: components/kyma-environment-broker/internal/config/validator.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+const requiredFields = "additional-components"
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/config/validator.go:11]: add a comment that it is a comma separated list of required fields

**Comment by piotrmiskiewicz** [comment]
please, rebase with main

File: components/kyma-environment-broker/internal/config/reader_test.go (hunk 1 above)
Comment (szwedm) [reply components/kyma-environment-broker/internal/config/reader_test.go:199]: I will rename this fake client to "failingK8sClient"

File: components/kyma-environment-broker/internal/config/validator.go (hunk 2 above)
Comment (szwedm) [reply components/kyma-environment-broker/internal/config/validator.go:11]: good idea, I will add appropriate comment

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**